docs/features/source-generators.cookbook.md

```diff
@@ -212,6 +212,58 @@ public partial class {userClass.Identifier}
 }
 ```
 
+### Issue Diagnostics
+
+**User Scenario:** As a generator author I want to be able to add diagnostics to the users compilation.
+
+**Solution:** Diagnostics can be added to the compilation via `SourceGeneratorContext.ReportDiagnostic()`. These can be in response to the content of the users compilation:
+for instance if the generator is expecting a well formed `AdditionalFile` but can not parse it, the generator could emit a warning notifying the user that generation can not proceed.
+
+For code-based issues, the generator author should also consider implementing a [diagnostic analyzer](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2019) that identifies the problem, and offers a code-fix to resolve it.
+
+**Example:**
+
+```csharp
+[Generator]
+public class MyXmlGenerator : ISourceGenerator
+{
+
+    private static readonly DiagnosticDescriptor InvalidXmlWarning = new DiagnosticDescriptor(id: "MYXMLGEN001",
```

Comment:
>MYXMLGEN [](start = 99, length = 8)

just curious, are some diagnostic prefixes disallowed as we do in analyzers in general? (for example "CS")

Reply:
As far as I know, we don't actually disallow prefixes, but we do have an analyzer that warns it shouldn't be used. I've confirmed that the analyzer issues the warnings for the generator in the above case.

@mavasani to confirm the expected behavior.

Reply:
Yep, see https://github.com/dotnet/roslyn-analyzers/issues/1727 and https://github.com/dotnet/roslyn/issues/25748 for more context. We attempted to restrict prefixes in the compiler, but that caused a breaking change, so we reverted it and instead implemented an analyzer that warns when using well-known/reserved prefixes such as CS, BC or CA.